src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,23 +38,34 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String latString = ""; 
+    String lngString = "";
+    String product = "";
+    
     try {
-    String latString = request.getParameter("lat");
-    String lngString = request.getParameter("lng");
-    double lat = convertToDouble(latString);
-    double lng = convertToDouble(lngString);
-    MapLocation userLocation = new MapLocation(lat, lng);
-    smallCityService.createUserServiceWithGeolocation(userLocation);
-    } catch(NullPointerException e) {
-      String zip = request.getParameter("zipCode");
-      smallCityService.createUserServiceWithZip(zip);
+      latString = request.getParameter("lat");
+      lngString = request.getParameter("lng");
+      product = request.getParameter("product");
+    } catch (NullPointerException e) {
       LOGGER.warning(e.getMessage() 
            + "Unable to geolocate user, zipCode entered instead.");
+    } finally {
```

Comment:
I don't think you gain anything with a finally block here. Here is a great resource on it: https://www.geeksforgeeks.org/flow-control-in-try-catch-finally-in-java/

Reply:
Yes you're right - I removed it.